diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/EmptyBlock.kt

```diff
@@ -45,38 +33,32 @@ class EmptyBlock : Rule("empty-block-structure") {
         isFixMode = autoCorrect
 
         val configuration = EmptyBlockStyleConfiguration(
-                configRules.getRuleConfig(BRACES_BLOCK_STRUCTURE_ERROR)?.configuration ?: mapOf()
+                configRules.getRuleConfig(EMPTY_BLOCK_STRUCTURE_ERROR)?.configuration ?: mapOf()
         )
         searchNode(node, configuration)
     }
 
     private fun searchNode(node: ASTNode, configuration: EmptyBlockStyleConfiguration) {
-        val newNode = when (node.elementType) {
-            THEN, ELSE -> node.findChildByType(BLOCK)?.findChildByType(LBRACE)!!.treeNext.treeParent
-            WHEN -> node.findChildByType(LBRACE)!!.treeNext.treeParent
-            FOR, WHILE, DO_WHILE -> node.findChildByType(BODY)?.findChildByType(BLOCK)?.findChildByType(LBRACE)!!.treeNext.treeParent
-            CLASS, OBJECT_DECLARATION -> node.findChildByType(CLASS_BODY)!!.findChildByType(LBRACE)!!.treeNext.treeParent
-            else -> if (node.hasChildOfType(BLOCK)) node.findChildByType(BLOCK)?.findChildByType(LBRACE)!!.treeNext.treeParent else null
-        } ?: return
+        val newNode = node.findLBrace()?.treeParent ?: return
         checkEmptyBlock(newNode, configuration)
     }
 
     private fun checkEmptyBlock(node: ASTNode, configuration: EmptyBlockStyleConfiguration) {
         if (node.treeParent.findChildByType(MODIFIER_LIST)?.findChildByType(OVERRIDE_KEYWORD) != null) return
         if (node.isBlockEmpty()) {
-            BRACES_BLOCK_STRUCTURE_ERROR.warnAndFix(configRules, emitWarn, isFixMode, "there can't be empty blocks in multi blocks",
+            EMPTY_BLOCK_STRUCTURE_ERROR.warnAndFix(configRules, emitWarn, isFixMode, "there can't be empty blocks in multi blocks",
```

Comment:
I kind of lost logic here: you warn if empty block is used, and then also warn if it is formatted incorrectly. How should user fix these warnings?